### PR TITLE
Qiskit version 0.14

### PR DESCRIPTION
**Pull request by antalszava**
With Qiskit's new release, certain parts of PL-Qiskit need to be modified.

This PR further includes minor changes related to the [series of creating new ``QNodes``](https://github.com/XanaduAI/pennylane/pull/440).

An issue has been opened for Qiskit, as a wide range of tests is failing due to a modification on their end: https://github.com/Qiskit/qiskit-terra/issues/3619

### CODE REVIEW

File: pennylane_qiskit/qiskit_device.py [hunk 1]
```diff
@@ -205,6 +205,9 @@ def apply(self, operation, wires, par):
 
         if operation == "QubitStateVector":
 
+            if self.backend_name == "unitary_simulator":
+                raise QuantumFunctionError("The QubitStateVector operation is not supported on the unitary simulator backend.")
+
```
Comment (antalszava) [inline pennylane_qiskit/qiskit_device.py]: These lines were added as per this issue: https://github.com/Qiskit/qiskit-aer/issues/497

File: pennylane_qiskit/qiskit_device.py [hunk 2]
```diff
@@ -315,7 +318,8 @@ def rotate_basis(self, obs, wires, par):
     def pre_measure(self):
         for e in self.obs_queue:
             # Add unitaries if a different expectation value is given
-            if hasattr(e, "return_type") and e.return_type == Sample:
+            # Exclude unitary_simulator as it does not support memory=True
+            if hasattr(e, "return_type") and e.return_type == Sample and self.backend_name != 'unitary_simulator':
```
Comment (antalszava) [inline pennylane_qiskit/qiskit_device.py]: "Backend support for memory=True now checked when that kwarg is passed. QiskitError results if not supported." - [from the release notes of version 0.14.0](https://qiskit.org/documentation/release_notes.html)

File: tests/test_apply.py [hunk 3]
```diff
@@ -68,6 +68,10 @@ class TestStateApply:
     def test_qubit_state_vector(self, init_state, device, tol):
         """Test PauliX application"""
         dev = device(1)
+
+        if dev.backend_name == "unitary_simulator":
+            pytest.skip("Test only runs for backends that are not the unitary simulator.")
```
Comment (josh146) [inline tests/test_apply.py]: Looks good! For maintainablity, I would suggest moving this into a fixture (e.g., `skip_unitary`), and adding it to function arguments where it should be skipped.

File: requirements.txt [hunk 4]
```diff
@@ -1,3 +1,4 @@
 qiskit>=0.12
 PennyLane>=0.7.0
 numpy
+networkx==2.3
```
Comment (antalszava) [inline requirements.txt]: Note to self: "Pinning networkx version to 2.3 such that tests pass for Python 3.5 as required by qiskit-terra"

**Pull request merged**